### PR TITLE
Reset style props set to null, undefined or false

### PR DESCRIPTION
**Pull request by linde12**
To be compatible with React we have to empty style props
when the value is null, undefined or false.

### CODE REVIEW

**Comment by linde12** [comment]
@developit I noticed that, in MaterialUI, after i hovered buttons the "hovered" styles were still applied. After some research i noticed that the style-props weren't cleared when setting a property to false, undefined or null like this:

`this.setState({backgroundColor: this.state.isHovering && 'gray'});`

This would become falsy when isHovering is false and so we have to remove the already applied `backgroundColor`

**Comment by linde12** [comment]
I'll fix the spaces -> tabs tomorrow :-)

File: src/dom/index.js [hunk 1]
```diff
@@ -40,7 +40,8 @@ export function setAccessor(node, name, old, value, isSvg) {
 		}
 		if (value && typeof value==='object') {
 			if (!isString(old)) {
-				for (let i in old) if (!(i in value)) node.style[i] = '';
+				for (let i in old)
+					if (!(i in value) || value[i] == null || value[i] === false) node.style[i] = '';
```
Comment (developit) [inline src/dom/index.js]: Can skip the `in` check now I think

Comment (linde12) [inline src/dom/index.js]: Done! :smile:

**Comment by developit** [comment]
@linde12 sitting on this for a bit because I think there might be a way to roll the empty check into the second iteration pass. I'll let you know if I think of it.

**Comment by linde12** [comment]
@developit Okay, great! :-)

**Comment by linde12** [comment]
@developit Should we resolve the conflicts? It looks like the diff is only the way we check if `old` is a string or not

**Comment by developit** [comment]
I think we actually need to move the `false` handling to L48, since the check should be on the new value ("if the value is now false, remove the property"):
https://github.com/developit/preact/pull/395/files#diff-4935bb00e75b7854383877cbfd9d770fR48

```js
node.style[i] = typeof value[i]==='number' && !NON_DIMENSION_PROPS[i] ? (value[i]+'px') : value[i]===false ? '' : value[i];
```

**Comment by linde12** [comment]
Ah, i think you're correct! If a value is falsy during the first render the style would, with the current code, be set to false. But if we always check the new properties we will avoid the problem i think.

**Comment by paranoidjk** [comment]
👍  +1 for this, when will this get merged?

**Comment by linde12** [comment]
Looks like a similar issue was merged on master. Can we close this?

**Pull request closed without merging**